pywr/parameters/_parameters.pyx

```diff
@@ -623,7 +659,7 @@ cdef class IndexedArrayParameter(Parameter):
         for p in params:
             if not isinstance(p, Parameter):
                 from pywr.parameters import ConstantParameter
-                p = ConstantParameter(p)
+                p = ConstantParameter(model, p)
             self.params.append(p)
 
         for param in params:
```

Comment:
This should be `for param in self.params`. It will miss any that are coerced to a `ConstantParameter` above.